iceoryx_dust/test/moduletests/test_cxx_span.cpp

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_dust/cxx/span.hpp"
+#include "iceoryx_hoofs/cxx/vector.hpp"
+#include "test.hpp"
+
+#include <vector>
+
+namespace
+{
+using namespace ::testing;
+using namespace iox::cxx;
+
+class span_test : public Test
+{
+  public:
+    void SetUp() override
+    {
+    }
+
+    void TearDown() override
+    {
+    }
+};
+
+TEST_F(span_test, NewSpanCreatedWithTheDefaultConstructor)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6a0589e2-b2fb-4f81-8082-6d3ea411e659");
+    span<int> dyn_span;
+    EXPECT_EQ(nullptr, dyn_span.data());
+    EXPECT_EQ(0u, dyn_span.size());
+
+    constexpr span<int, 0> static_span;
+    static_assert(nullptr == static_span.data(), "");
+    static_assert(0u == static_span.size(), "");
+}
+
+TEST_F(span_test, NewSpanFromIterator)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "92de7ff5-d03e-41ae-bfdd-893892381b24");
+    constexpr int* kNull = nullptr;
+    constexpr span<int> empty_span(kNull, 0);
+    EXPECT_TRUE(empty_span.empty());
```

Comment:
This will end the test if not fulfilled and not continue to work on corrupted data.

```suggestion
    ASSERT_TRUE(empty_span.empty());
```

Reply:
Done.